### DISCUSSION TITLE
Update Marlin's firmware over WIFI with combination ESP8266 + ESP3D

### DISCUSSION THREAD

**Discussion by Surrizu** [luc-github/ESP3D#569] [Ideas]
Hello. I would like to make possible update the arduino firmware over WIFI.

My hardware combination is the following: Marlin 2.0 + RAMPS 1.4 + ESP8266(NodeMCU) + ESP3D

NodeMCU has only 4 wires connected to RAMPS --> Vcc / GND / Tx / Rx

BUT, I´m pretty sure that I´m missing something, because I believe I should connect any GPIO pin (of NodeMCU) to Arduino RESET pin. Then I think I could update Marlin firmware over wifi. 

Does anyone have tried this??

**Comment by luc-github** [comment]
it is not that simple -  you may want to try this lib https://github.com/ciminaghi/libdfu

and may be also https://github.com/esp8266/Arduino/tree/master/libraries/ESP8266AVRISP
https://github.com/esp8266/Arduino/blob/master/libraries/ESP8266AVRISP/examples/Arduino_Wifi_AVRISP/Arduino_Wifi_AVRISP.ino
